src/status_im/utils/ethereum/erc20.cljs

```diff
@@ -57,4 +62,128 @@
+            :from          (-> entry :topics second remove-padding)
+            :to            (-> entry :topics last remove-padding)
+            :value         (-> entry :data ethereum/hex->bignumber)
+            :type          direction
+
+            :confirmations (str (- block-number (-> entry :blockNumber ethereum/hex->int)))
+
+            :gas-price     "0"
+            :nonce         "0"
+            :data          "0x"
+
+            :gas-limit     "0"
+            ;; NOTE(goranjovic) - timestamp is mocked to the current time so that the transaction is shown at the
+            ;; top of transaction history list between the moment when transfer event was detected and actual
+            ;; timestamp was retrieved from block info.
+            :timestamp     (str (.getTime (cljs-time/now)))
+
+            :gas-used      "0"
+
+            ;; NOTE(goranjovic) - just a flag we need when we merge this entry with the existing entry in
+            ;; the app, e.g. transaction info with gas details, or a previous transfer entry with old
+            ;; confirmations count.
+            :transfer      true}])))
+
+(defn- response-handler [block-number chain direction error-fn success-fn]
+  (fn handle-response
+    ([response]
+     (let [{:keys [error result]} (parse-json response)]
+       (handle-response error result)))
+    ([error result]
+     (if error
+       (error-fn error)
+       (success-fn (parse-transaction-entry block-number chain direction result))))))
+
+;;
+;; Here we are querying event logs for Transfer events.
+;;
+;; The parameters are as follows:
+;; - address - token smart contract address
+;; - fromBlock - we need to specify it, since default is latest
+;; - topics[0] - hash code of the Transfer event signature
+;; - topics[1] - address of token sender with leading zeroes padding up to 32 bytes
+;; - topics[2] - address of token sender with leading zeroes padding up to 32 bytes
+;;
+
+(defn get-token-transfer-logs
+  ;; NOTE(goranjovic): here we cannot use web3 since events don't work with infura
+  [block-number network contracts direction address cb]
+  (let [chain (ethereum/network->chain-keyword network)
+        [from to] (if (= :inbound direction)
+                    [nil (ethereum/normalized-address address)]
+                    [(ethereum/normalized-address address) nil])
+        args {:jsonrpc "2.0"
+              :id      2
+              :method  constants/web3-get-logs
+              :params  [{:address   contracts
+                         :fromBlock "0x0"
+                         :topics    [constants/event-hash-transfer
+                                     (add-padding from)
+                                     (add-padding to)]}]}
+        payload (.stringify js/JSON (clj->js args))]
+    (status/call-web3-private payload
+                              (response-handler block-number chain direction js/alert cb))))
+
+(defn- handle-error [error]
+  (log/info (.stringify js/JSON error)))
```

Comment:
This will go unnoticed by end users. Is it expected?

Reply:
Yes, it's only supposed to appear in logs at the moment.